kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/formats/AvroFormatWriter.scala

```diff
@@ -31,7 +38,22 @@ import scala.util.Failure
 import scala.util.Success
 import scala.util.Try
 
-class AvroFormatWriter(outputStreamFn: () => S3OutputStream) extends S3FormatWriter with LazyLogging {
+class AvroFormatWriter(outputStreamFn: () => S3OutputStream)(implicit compressionCodec: CompressionCodec)
+    extends S3FormatWriter
+    with LazyLogging {
+
+  private val avroCompressionCodec: CodecFactory = {
+    compressionCodec match {
+      case CompressionCodec(UNCOMPRESSED, _)      => CodecFactory.nullCodec()
+      case CompressionCodec(SNAPPY, _)            => CodecFactory.snappyCodec()
+      case CompressionCodec(BZIP2, _)             => CodecFactory.bzip2Codec()
+      case CompressionCodec(ZSTD, Some(level))    => CodecFactory.zstandardCodec(level)
```

Comment:
I am not sure, but I think I saw some errors around compression if the required libs are not present, the underlying C libs.  Can we make sure we cover the compression in integration tests?

Reply:
This is correct.  I've added tests for the ones that I can easily, however some of the dependencies require obscure libraries/packages installed and I a testcontainers issue prevented me from getting through with this.  The coverage is mostly there though!